src/cli/file.py

```diff
@@ -1,7 +1,24 @@
 """
-Not implemented yet.
+Upload a local file and share its public URL.
+
+Usage: instantshare file [options] <filename>
+
+Options:
+  -h, --help            Print this information
+  --storage=<provider>  Overwrite the storage config parameter
 """
+from docopt import docopt
+
+import storage
 
 
 def main(argv):
-    print(__doc__)
+    args = docopt(__doc__, argv=argv)
+
+    hoster = args["--storage"]
+    file = args["<filename>"]
+
+    if hoster:
+        storage.upload_to(hoster, file)
```

Comment:
Any checks of that kind would need to be done inside the different storage modules. We don't really have proper error handling for that yet.